rundeckapp/grails-app/services/rundeck/services/NotificationService.groovy

```diff
@@ -283,6 +283,11 @@ public class NotificationService implements ApplicationContextAware{
                         if (configAttachLogInFile in ['true',true]) {
                             attachlog = true
                         }
+
+                        //for old versions support
+                        if(!attachlog && !attachlogbody){
```

Comment:
You could add a unit test for this to avoid that by mistake that code goes back in a future change.